store.go

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package rate
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+type entry struct {
+	key   string
```

Comment:
do we want to create a key type and enforce any constraints on it?

Reply:
Do you have any constraints in mind? I don't see any that make sense for the library to enforce.

Reply:
can a key be any string, or does it have a structure of known segments?

Reply:
While there is a structure for this key that the [expireableStore](https://github.com/hashicorp/go-rate/blob/4ad2b632dfc2ef0a8f6f00d9556079164213b4ea/store.go#L121) uses, I think from the point of view of the entry it is opaque. There is no need to get that structure back from the key. If/when we look at remote persistence of Quotas there might be a need for it, but I think I would rather wait until then and refactor this if necessary.